### PR TITLE
Funding of market

### PR DESCRIPTION
**Pull request by kongzii**
## Summary by CodeRabbit

- **New Features**
	- Enhanced funding and fund removal capabilities for Omen markets in the prediction market tooling.
- **Tests**
	- Expanded test coverage for market creation, funding, and fund removal operations.

### CODE REVIEW

File: tests/markets/test_omen.py [hunk 1]
```diff
@@ -92,6 +95,35 @@ def test_omen_create_market() -> None:
...
+    remove_fund = omen_outcome_type(xdai_to_wei(xdai_type(0.01)))
+    keys = APIKeys()
+    omen_fund_market_tx(
+        market=market,
+        funds=funds,
+        from_address=keys.bet_from_address,
+        from_private_key=keys.bet_from_private_key,
+        auto_deposit=True,
+    )
+    time.sleep(10)  # Wait for the transaction to be mined.
```
Comment (evangriffiths) [inline tests/markets/test_omen.py:117]: Is there something we could poll to see if it's ready?
e.g. 
```
def wait_until_nonce_has_updated(current_nonce: Nonce) -> None:
    while nonce_has_not_updated(current_nonce):
        time.sleep(0.1)
```

Comment (kongzii) [reply tests/markets/test_omen.py:117]: We are actually doing something like that in https://github.com/gnosis/prediction-market-agent-tooling/blob/2b2cfc36a20c32cb5a6325a9e6e6cdbe06e257f3/prediction_market_agent_tooling/tools/web3_utils.py#L130.

I thought that that should be it. But somehow it isn't enough (maybe free RPC's node is behind a bit?).

Comment (kongzii) [reply tests/markets/test_omen.py:117]: I'll try the nonce trick

**Review by evangriffiths — APPROVED** [review]

File: tests/markets/test_omen.py (hunk 1 above)
Comment (kongzii) [reply tests/markets/test_omen.py]: Seems like it works!! 🎉

**Pull request merged**